### PR TITLE
GraphQL Guard on Legacy Datastore Invocations

### PR DESCRIPTION
**Pull request by keyallis**
## Description

A detailed explanation of the changes in your PR.

Feel free to remove this section if it is overkill for your PR, and the title of your PR is sufficiently descriptive.

## Checklist
- [ ] Investigated and inspected CI test results
- [ ] Unit test and regression tests added
- [ ] Evaluated and added CHANGELOG entry if required
- [ ] Determined and documented upgrade steps
- [ ] Documented user facing changes (create PR based on [openshift/openshift-docs](https://github.com/openshift/openshift-docs) and merge into [rhacs-docs](https://github.com/openshift/openshift-docs/tree/rhacs-docs))

If any of these don't apply, please comment below.

## Testing Performed

TODO(replace-me)
Use this space to explain how you tested your PR, or, if you didn't test it, why
you did not do so. Valid reasons include, for example, "CI is sufficient",
"No testable changes". Feel free to attach JSON snippets, curl commands,
screenshots.

In addition to reviewing your code, reviewers **must** also review your testing
instructions and make sure they are sufficient.

### CODE REVIEW

File: central/graphql/resolvers/components_v2.go [hunk 1]
```diff
@@ -111,6 +111,9 @@ func (resolver *Resolver) componentsV2Query(ctx context.Context, query *v1.Query
 }
 
 func (resolver *Resolver) imageComponentDataStoreQuery(ctx context.Context, args IDQuery) (*imageComponentResolver, error) {
+	if features.PostgresDatastore.Enabled() {
```
Comment (md2119) [inline central/graphql/resolvers/components_v2.go:114]: This is fine here, assuming you are planning a clean code split on components and that new code does not in any way invoke this func.

**Review by md2119 — APPROVED** [review]

File: central/graphql/resolvers/components_v2.go (hunk 1 above)
Comment (keyallis) [reply central/graphql/resolvers/components_v2.go:114]: That's the plan, if it ends up being relevant I'll tweak it in the PR for image component support

**Comment by keyallis** [comment]
`/retest`

**Pull request merged**